### DISCUSSION TITLE
argo-workflows/tls/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#10185] [General]
# argo-workflows/tls/

https://argoproj.github.io/argo-workflows/tls/

**Comment by liuyix** [comment]
I stronly discourage default encrypt configuration enabled. Reasons: 1) cert is self-signed which is not friendly both dev and prod. For dev, curl / client / browser must set ignore or skip and do serveral extra workarounds to make things work (as the document says). 
For truly production deploy, the documentation simply says 'Run your HTTPS proxy in front of the Argo Server. You'll need to set-up your certificates (this is out of scope of this documentation).' which is insufficient and useless. 
In my usage scene, `argo-server` is behind the scene and not open at all, therefore encryption benefits no one.

**Reply by hholst80** [reply]
I could not agree more. If I need to expose Argo over a publicly accessible network I will setup this. Using a self signed cert for a browser accessible UI is an anti-pattern of security and makes users custom to ignoring TLS warnings.
